snow/validators/gvalidators/validator_state_client.go

```diff
@@ -86,9 +86,14 @@ func (c *Client) GetValidatorSet(
 			}
 		}
 		vdrs[nodeID] = &validators.GetValidatorOutput{
-			NodeID:    nodeID,
-			PublicKey: publicKey,
-			Weight:    validator.Weight,
+			NodeID: nodeID,
+			// As an optimization, we pre-populate the cached deserialized key
+			// since it was given to us off gRPC.
```

Comment:
I'm wondering if we should make is so that we perform `Deserialize` on demand. (Rather than deserializing every provided key)

Because the warp verifier only need the public keys that were actually used in the signature creation, we should be able to avoid `Deserialize`ing a number of keys.

Reply:
Hmm yeah that's definitely an interesting idea. We should definitely be able to lazy-deserialize as needed. Will add this to this PR.